src/uct/api/uct_def.h

```diff
@@ -261,6 +263,68 @@ typedef void (*uct_unpack_callback_t)(void *arg, const void *data, size_t length
 
 
 /**
+ * @ingroup UCT_RESOURCE
+ * @brief Callback to process an incoming connection request message on the server
+ *        side.
+ *
+ * This callback routine will be invoked on the server side upon receiving an
+ * incoming connection request.
+ * Incoming data is placed inside the conn_priv_data buffer and outgoing data should
+ * be filled inside this function and placed in the reply_priv_data.
```

Comment:
What are requirements for the callback ? I guess it has to be thread safe ? Can it invoke other UCT calls ?
The same relevant for all the rest